ansible/roles/test/files/ptftests/vnet_vxlan.py

```diff
@@ -0,0 +1,311 @@
+            print
+
+    def FromVM(self, test):
+        rv = True
+        if test['vlan'] != 0:
+            tagged = True
+        else:
+            tagged = False
+        for n in self.net_ports:
+            pkt = simple_tcp_packet(
+                eth_dst=self.dut_mac,
+                eth_src=self.random_mac,
+                ip_dst=test['src'],
+                ip_src=test['dst'],
+                ip_id=108,
+                ip_ttl=64)
+            udp_sport = 1234 # Use entropy_hash(pkt)
+            vxlan_pkt = simple_vxlan_packet(
+                eth_dst=self.dut_mac,
+                eth_src=self.random_mac,
+                ip_id=0,
+                ip_src=test['host'],
+                ip_dst=self.loopback_ip,
+                ip_ttl=64,
+                udp_sport=udp_sport,
+                vxlan_vni=test['vni'],
+                with_udp_chksum=False,
+                inner_frame=pkt)
+            exp_pkt = simple_tcp_packet(
+                eth_src=self.dut_mac,
+                eth_dst=self.ptf_mac_addrs['eth%d' % test['port']],
+                dl_vlan_enable=tagged,
+                vlan_vid=test['vlan'],
+                ip_dst=test['src'],
+                ip_src=test['dst'],
+                ip_id=108,
+                ip_ttl=63)
+            send_packet(self, n, str(vxlan_pkt))
+
+            log_str = "Sending packet from port " + str(n) + " to " + test['src']
+            logging.info(log_str)
+            print log_str
+
+            log_str = "Expecing packet on " + str("eth%d" % test['port']) + " from " + test['dst']
+            logging.info(log_str)
+            print log_str
+
+            verify_packet(self, exp_pkt, test['port'])
+
+
+    def FromServer(self, test):
+        rv = True
+        try:
+            if test['vlan'] != 0:
+                tagged = True
+            else:
+                tagged = False
+            pkt = simple_tcp_packet(
+                eth_dst=self.dut_mac,
+                eth_src=self.ptf_mac_addrs['eth%d' % test['port']],
+                dl_vlan_enable=tagged,
+                vlan_vid=test['vlan'],
+                ip_dst=test['dst'],
+                ip_src=test['src'],
+                ip_id=105,
+                ip_ttl=64)
+            exp_pkt = simple_tcp_packet(
+                eth_dst=test['mac'],
+                eth_src=self.dut_mac,
+                ip_dst=test['dst'],
+                ip_src=test['src'],
+                ip_id=105,
+                ip_ttl=63)
+            udp_sport = 1234 # Use entropy_hash(pkt)
+            encap_pkt = simple_vxlan_packet(
+                eth_src=self.dut_mac,
+                eth_dst=self.random_mac,
+                ip_id=0,
+                ip_src=self.loopback_ip,
+                ip_dst=test['host'],
+                ip_ttl=64,
+                udp_sport=udp_sport,
+                with_udp_chksum=False,
+                vxlan_vni=test['vni'],
+                inner_frame=exp_pkt)
```

Comment:
Add encap_pkt[IP].flags = 0x2 here because BF set DF bit in IP header (due to increased header size)